doc/api/n-api.md

```diff
@@ -3782,6 +3804,178 @@ NAPI_EXTERN napi_status napi_get_uv_event_loop(napi_env env,
+non-blockingly, returning `napi_queue_full` if the data was not successfully
+added to the queue. If set to `true`, the API blocks until space becomes
+available in the queue. `napi_call_threadsafe_function()` never blocks if the
+thread-safe function was created with a maximum queue size of 0.
+
+The callback given via the `call_js_cb` parameter is invoked on the main thread
+for each call to `napi_call_threadsafe_function()`. If such a callback is not
+given, a default callback will be used, and the resulting JavaScript call will
+have no arguments. The `call_js_cb` callback receives the JavaScript function to
+call as a `napi_value` in its parameters, as well as the `void*` hint pointer
+used when creating the `napi_threadsafe_function`, and the next data pointer
+that was created by one of the secondary threads. The callback can then use an
+API such as `napi_call_function()` to call into JavaScript.
+
+*Note*: It is not necessary to call into JavaScript via `napi_make_callback()`
+because N-API runs `call_js_cb` in a context appropriate for callbacks.
+
+`napi_threadsafe_function` objects are reference counted. Thus, in addition to
+specifying an initial reference count upon creation, the APIs
+`napi_ref_threadsafe_function` and `napi_unref_threadsafe_function` can be
+called to increase resp. decrease the reference count. The reference count is
+meant to coincide with the number of threads, including the main thread, making
+use of the `napi_threadsafe_function`.
+
+`napi_threadsafe_function` objects are destroyed when every thread which uses
+the object has called `napi_unref_threadsafe_function()`. The queue is emptied
+before the `napi_threadsafe_function` is destroyed. It is important that
+`napi_unref_threadsafe_function()` be the last API call made in conjunction
+with a given `napi_threadsafe_function`, because after the call completes, there
+is no guarantee that the `napi_threadsafe_function` is still allocated. Data
+associated with the `napi_threadsafe_function` can be freed in its
+`napi_finalize` callback which is passed to `napi_create_threadsafe_function()`.
+
+Once the reference count of a `napi_threadsafe_function` reaches zero, it can no
+longer be incremented. In fact, all APIs associated with it will return an error
+value of `napi_closing`.
+
+### napi_create_threadsafe_function
+
+<!-- YAML
+added: REPLACEME
+-->
+```C
+NAPI_EXTERN napi_status
+napi_create_threadsafe_function(napi_env env,
+                                napi_value func,
+                                size_t max_queue_size,
+                                size_t initial_refcount,
+                                void *thread_finalize_data,
+                                napi_finalize thread_finalize_cb,
+                                void *hint,
+                                napi_threadsafe_function_call_js call_js_cb,
+                                napi_threadsafe_function* result);
```

Comment:
nit: We should be consistent with pointer style. I’d prefer left-leaning pointers because that’s what we use elsewhere in Node.js.